auth.py

```diff
@@ -0,0 +1,58 @@
+from flask_login import LoginManager, login_user
+from flask import redirect
+import requests
+from TodoUser import TodoUser
+from oauthlib.oauth2 import WebApplicationClient
+import random, string
+import os
+
+OAUTH_ID = os.environ.get('OAUTH_ID')
+OAUTH_SECRET = os.environ.get('OAUTH_SECRET')
+SECRET_KEY = os.environ.get('secret_key')
+
+
+def init_auth(app):
+    login_mgr = LoginManager()
+
+    app.secret_key = SECRET_KEY
+
+    @login_mgr.unauthorized_handler
+    def unauthorised():
+        state = ''.join(random.choices(string.ascii_lowercase, k=16))
+        auth_client = WebApplicationClient(OAUTH_ID)
+        auth_uri = auth_client.prepare_request_uri(
+            'https://github.com/login/oauth/authorize',
+            redirect_uri='http://127.0.0.1:5000/login/callback',
```

Comment:
You can just omit the redirect_uri and rely on the OAuth app's config in GitHub.

This hardcoded value is wrong for other environments, so if you do want to include it, it'll have to be an env var.